substrate/client/consensus/babe/src/lib.rs

```diff
@@ -1413,6 +1366,77 @@ where
 		let hash = block.post_hash();
 		let number = *block.header.number();
 		let info = self.client.info();
+		let parent_hash = *block.header.parent_hash();
+
+		let create_inherent_data_providers = self
+			.create_inherent_data_providers
+			.create_inherent_data_providers(parent_hash, ())
+			.await?;
+
+		let slot_now = create_inherent_data_providers.slot();
+
+		let babe_pre_digest = find_pre_digest::<Block>(&block.header)
+			.map_err(|e| ConsensusError::Other(Box::new(e)))?;
+		let slot = babe_pre_digest.slot();
+
+		// Check inherents.
+		if let Some(inner_body) = block.body {
+			let new_block = Block::new(block.header.clone(), inner_body);
+			if !block.state_action.skip_execution_checks() {
+				// if the body is passed through and the block was executed,
+				// we need to use the runtime to check that the internally-set
+				// timestamp in the inherents actually matches the slot set in the seal.
+				let mut inherent_data = create_inherent_data_providers
+					.create_inherent_data()
+					.await
+					.map_err(|e| ConsensusError::Other(Box::new(e)))?;
+				inherent_data.babe_replace_inherent_data(slot);
+				self.check_inherents(
+					new_block.clone(),
+					parent_hash,
+					inherent_data,
+					create_inherent_data_providers,
+				)
+				.await?;
+			}
+			let (_, inner_body) = new_block.deconstruct();
+			block.body = Some(inner_body);
+		}
+
+		// Check for equivocation and report it to the runtime if needed.
+		let author = {
+			let parent_header_metadata = self
+				.client
+				.header_metadata(parent_hash)
+				.map_err(|e| ConsensusError::Other(Box::new(e)))?;
+			let epoch_changes = self.epoch_changes.shared_data();
+			let epoch_descriptor = epoch_changes
+				.epoch_descriptor_for_child_of(
+					descendent_query(&*self.client),
+					&parent_hash,
+					parent_header_metadata.number,
```

Comment:
```suggestion
					number - 1,
```

Is easier (I know you copied this from before)